snuba/clickhouse/translators/snuba/mappers.py

```diff
@@ -44,14 +47,77 @@ def attempt_map(
             and expression.table_name == self.from_table_name
         ):
             return ColumnExpr(
-                alias=expression.alias,
+                alias=expression.alias
+                or qualified_column(self.from_col_name, self.from_table_name or ""),
```

Comment:
This is the bug I was talking about in the description. If the alias is empty (like most of the times) we should pick up the original column name, that is what the user expects. 
This should not be needed here. We should just apply, at parsing time, the alias to all expressions like we do for SubscriptableReference (https://github.com/getsentry/snuba/blob/master/snuba/query/parser/strings.py#L25-L31).
Though that alias is not applied consistently yet, thus we need to cover for empty alias here.
I will fix the parser soon. Then we will always have expression.alias populated and we won't care about the column name anymore and we will be able to remove this logic.

Small remark on the qualified_column function. it calls the table `alias`, that's confusing now but it comes from the joins time when the table name was actually the table alias in the joinn.

Reply:
Makes sense. Unless you're extremely certain that is going to happen soon, this seems like one of those things that should be noted in inline to avoid confusion later.

> Small remark on the qualified_column function. it calls the table `alias`, that's confusing now but it comes from the joins time when the table name was actually the table alias in the joinn.

I agree, it should be updated (not necessarily as part of this change.)

Reply:
"Unless you're extremely certain that is going to happen soon

I am planning to do it this week, it would be hard to keep this around for long.